ui-tests/src/test/java/plugins/WarningsNextGenerationPluginTest.java

```diff
@@ -0,0 +1,621 @@
+package plugins;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.Test;
+
+import com.google.inject.Inject;
+
+import org.jenkinsci.test.acceptance.docker.DockerContainerHolder;
+import org.jenkinsci.test.acceptance.docker.fixtures.JavaGitContainer;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithCredentials;
+import org.jenkinsci.test.acceptance.junit.WithDocker;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.plugins.maven.MavenInstallation;
+import org.jenkinsci.test.acceptance.plugins.maven.MavenModuleSet;
+import org.jenkinsci.test.acceptance.plugins.ssh_slaves.SshSlaveLauncher;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.AnalysisResult;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.AnalysisResult.Tab;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.AnalysisSummary;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.AnalysisSummary.InfoType;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.ConsoleLogView;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.DefaultWarningsTableRow;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.DetailsTableRow;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.DryIssuesTableRow;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.InfoView;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.IssuesRecorder;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.IssuesRecorder.QualityGateBuildResult;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.IssuesRecorder.QualityGateType;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.IssuesTable;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.ScrollerUtil;
+import org.jenkinsci.test.acceptance.plugins.warnings_ng.SourceView;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.DumbSlave;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.Job;
+import org.jenkinsci.test.acceptance.po.Slave;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+
+import static org.jenkinsci.test.acceptance.plugins.warnings_ng.Assertions.*;
+
+/**
+ * Acceptance tests for the Warnings Next Generation Plugin.
+ *
+ * @author Frank Christian Geyer
+ * @author Ullrich Hafner
+ * @author Manuel Hampp
+ * @author Anna-Maria Hardi
+ * @author Elvira Hauer
+ * @author Deniz Mardin
+ * @author Stephan Plöderl
+ * @author Alexander Praegla
+ * @author Michaela Reitschuster
+ * @author Arne Schöntag
+ * @author Alexandra Wenzel
+ * @author Nikolai Wohlgemuth
+ * @author Florian Hageneder
+ * @author Veronika Zwickenpflug
+ */
+@WithPlugins("warnings-ng")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class Fan-Out Complexity is 31 (max allowed is 25).](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)